src/libAtomVM/nifs.c

```diff
@@ -1222,13 +1222,17 @@ static term do_spawn(Context *ctx, Context *new_ctx, size_t arity, size_t n_free
     term link_term = interop_proplist_get_value(opts_term, LINK_ATOM);
     term monitor_term = interop_proplist_get_value(opts_term, MONITOR_ATOM);
     term heap_growth_strategy = interop_proplist_get_value_default(opts_term, ATOMVM_HEAP_GROWTH_ATOM, BOUNDED_FREE_ATOM);
-    term request_term = interop_proplist_get_value(opts_term, REQUEST_ATOM);
+    term request_term = interop_proplist_get_value_default(opts_term, REQUEST_ATOM, UNDEFINED_ATOM);
     term group_leader;
+    bool valid_request = false;
```

Comment:
Testing request == UNDEFINED_ATOM and this bool is probably the same, but ok.

Reply:
that was my first idea, but in case one day we'll accept a further type, we might introduce a bug by mistake. TBH that function is quite big, and hard to maintain.